### ISSUE TITLE
Feature request: scoped shadow_recode variants

### ISSUE THREAD

**Issue by benifex** [njtierney/naniar#274]
Pretty self-explanatory I guess .. ? 

These would be terrific when there are large numbers of variables which by their nature will have a particular type of missingness, e.g. a questionnaire with filter questions that lead to other questions being skipped (or, in pseudocode, where Q1 is the filter, IF Q1 = "no", GOTO Q5; ELSE GOTO Q2). 

This is related to #249 - a superset of it I suppose.

I also note #262 below about the use of across() instead of scoped variants, which may impinge on this request. But whichever implementation is used, it'd be nice to have a parsimonious way to do this which is baked in. The familiar ~_all, ~_at and _~if makes things easier too.

**Comment by njtierney** [comment]
Yup, this is a great feature request, and something that I've wanted to implement, but it's great to have someone who will be there to help kick the wheels, so thank you for posting!

It'll be a few weeks as I'm taking a holiday but will return to this.

Thanks for posting the issue

**Comment by nathanj3** [comment]
I just discovered naniar, and I'm excited to see it progress.

Right now, the lack of shadow_recode variants is my biggest pain point -- in a Qualtrics survey, for example, I want to be able to globally label 9999 as "Skipped". Something like 

```
data_clean |>
  bind_shadow() |>
  recode_shadow(
    across(
      everything(),
      \(x) .where(x == 9999 ~ 'skipped')
    )
  )
```

would be incredible.
